mateweather/mateweather-dialog.c

```diff
@@ -502,6 +502,35 @@ static PangoFontDescription* get_system_monospace_font(void)
     return desc;
 }
 
+static void
+override_widget_font (GtkWidget   *widget,
+                      const gchar *font)
+{
+    gchar          *css;
+    GtkCssProvider *provider;
+    gchar          *tempsize;
+
+    provider = gtk_css_provider_new ();
+    tempsize = g_strdup (font);
+
+    g_strreverse (tempsize);
+    g_strcanon (tempsize, "1234567890", '\0');
+    g_strreverse (tempsize);
+
+    gchar tempfont [strlen (font)];
+    strcpy (tempfont, font);
+    tempfont [strlen (font) - strlen (tempsize)] = 0;
+
+    css = g_strdup_printf ("* { font-family: %s; font-size: %spt; }", tempfont, tempsize);
```

Comment:
Why not use `pango_font_description_get_family` and `pango_font_description_get_size`?

Reply:
Why if this code works?

Reply:
Feel free to open an alternative PR

Reply:
And there  is one more memory leak. I don't see where you free `tempsize`.

Reply:
Why should I do that? I just pointed out that there are problems...